cmd/ta/main.go

```diff
@@ -2,275 +2,69 @@ package main
 
 import (
 	"bytes"
-	"crypto/rand"
-	"encoding/hex"
 	"fmt"
 	"html/template"
 	"log"
-	"net/http"
 	"os"
 
-	btcec "github.com/btcsuite/btcd/btcec/v2"
-	"github.com/decred/dcrd/dcrec/secp256k1/v4"
-	"github.com/gin-gonic/gin"
 	"github.com/rddl-network/ta_attest/config"
-	"github.com/spf13/viper"
+	"github.com/rddl-network/ta_attest/service"
 
-	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/planetmint/planetmint-go/app"
-	"github.com/planetmint/planetmint-go/lib"
-	machinetypes "github.com/planetmint/planetmint-go/x/machine/types"
+	"github.com/spf13/viper"
 )
 
-func LoadConfig(path string) (v *viper.Viper, err error) {
-	v = viper.New()
+func loadConfig(path string) (cfg *config.Config, err error) {
+	v := viper.New()
 	v.AddConfigPath(path)
 	v.SetConfigName("app")
 	v.SetConfigType("env")
-
 	v.AutomaticEnv()
 
 	err = v.ReadInConfig()
 	if err == nil {
+		cfg = config.GetConfig()
+		cfg.ServiceBind = v.GetString("SERVICE_BIND")
+		cfg.ServicePort = v.GetInt("SERVICE_PORT")
+		cfg.PlanetmintActor = v.GetString("PLANETMINT_ACTOR")
+		cfg.PlanetmintChainID = v.GetString("PLANETMINT_CHAIN_ID")
+		cfg.FirmwareESP32 = v.GetString("FIRMWARE_ESP32")
+		cfg.FirmwareESP32C3 = v.GetString("FIRMWARE_ESP32C3")
+		cfg.TestnetMode = v.GetBool("TESTNET_MODE")
 		return
 	}
-	log.Println("no config file found.")
+	log.Println("no config file found")
 
-	tmpl := template.New("appConfigFileTemplate")
-	configTemplate, err := tmpl.Parse(config.DefaultConfigTemplate)
+	template := template.New("appConfigFileTemplate")
```

Comment:
I know that Go allows that but I'd prefer to leave the local variable named `tmpl`, so that it does not clash with the package name of `"html/template"`